packages/docsify-changelog-plugin/src/index.js

```diff
@@ -33,11 +33,213 @@ const plugin = (hook, vm) => {
+    // Place the setting inject into the CSS here
+    // We need to first build the entire settings object
+    var u = {
+      path: "changelog.md",
+      window: {
+        height: "60vh",
+        width: "400px",
+        portrait_height: "60vh",
+        portrait_width: "400px",
+        background: "#FFF",
+        scroll_track: "inherit",
+        scroll_button: "grey",
+        border: "rgba(0, 0, 0, 0.1)",
+        shadow: "rgba(0, 0, 0, .3)",
+        shadow_params: "0 0 34px 15px",
+        offset_right: "4vw"
+      },
+      button:{
+        notify: "orange",
+        text: "goldenrod",
+        text_hover: "darkorange",
+        text_transition: "1s",
+        background: "inherit",
+        background_hover: "grey",
+        background_transition: "1s",
+        brightness_hover: "100%",
+        position: "absolute",
+        offset_right: "0px"
+      },
+      header:{
+        auto_links: "none", 
+        border_edge: "rgba(0, 0, 0, 0.1)",
+        h1_text: "inherit",
+        h1_background: "rgba(0, 0, 0, 0.03)",
+        h1_size: "1.5rem",
+        h2_text: "inherit",
+        h2_background: "inherit",
+        h2_size: "2rem",
+        h3_text: "inherit",
+        h3_background: "inherit",
+        h3_size: "24px",
+        h4_text: "inherit",
+        h4_background: "inherit",
+        h4_size: "inherit"
+      },
+      text: {
+        all: "black",
+        strong: "inherit",
+        strong_background: "silver",
+        p: "darkred",
+        p_background: "inherit",
+        blockquote: "inherit",
+        blockquote_background: "inherit",
+        list: "inherit",
+        list_background: "inherit",
+        list_icon: `"\\f00c"` // accepts "caret", "angle", "chevron", "calendar", "bars", "check", or "arrow"
+      },
+      link: {
+        text: "inherit",
+        text_hover: "orange",
+        text_transition: "0.5s",
+        background: "inherit",
+        background_hover: "inherit",
+        background_transition: "0.5s"
+      }
+    };
+    var a = vm.config.changelog;
+    if (Object.isObject(a)) {
+      u.path = a.path || u.path;
+      if (Object.isObject(a.window)) {
+        u.window.height = a.window.height || u.window.height
+        , u.window.width = a.window.width || u.window.width
+        , u.window.portrait_height = a.window.portrait_height || u.window.portrait_height
+        , u.window.portrait_width = a.window.portrait_width || u.window.portrait_width
+        , u.window.background = a.window.background || u.window.background
+        , u.window.scroll_track = a.window.scroll_track || u.window.scroll_track
+        , u.window.scroll_button = a.window.scroll_button || u.window.scroll_button
+        , u.window.border = a.window.border || u.window.border
+        , u.window.shadow = a.window.shadow || u.window.shadow
+        , u.window.shadow_params = a.window.shadow_params || u.window.shadow_params
+        , u.window.offset_right = a.window.offset_right || u.window.offset_right;
```

Comment:
try simplifying these long codes

```js
u.window = Object.assign({ }, a.window, u.window)
```

Reply:
its your choice though, if you are comfortable with working in a current way feel free to do so 👍 
I will change them later

Reply:
I just began learning Javascript when I put together this website for my project, so I would consider myself still novice level when dealing with proper methods. So any advice on stylings and methods are welcome.

Reply:
Yes, I looked at the documentation, that is exactly what I needed. ty